### PR TITLE
Add sequential kinetic operator

### PR DESCRIPTION
**Pull request by jwnys**
@gpescia

### CODE REVIEW

File: netket/operator/_kinetic.py [hunk 1]
```diff
@@ -60,18 +63,22 @@ def __init__(
         hilbert: AbstractHilbert,
         mass: Union[float, list[float]],
         dtype: Optional[DType] = None,
+        method: str = "parallel",
     ):
         r"""Args:
         hilbert: The underlying Hilbert space on which the operator is defined
         mass: float if all masses are the same, list indicating the mass of each particle otherwise
         dtype: Data type of the matrix elements. Defaults to `np.float64`
+        method: Method to compute the kinetic energy: "sequential" or "parallel", where sequential uses less memory
```
Comment (gpescia) [inline netket/operator/_kinetic.py:72]: I would either rename these to "vmap" and "forloop" or something similar or say what is actually meant with 'sequential' and 'parallel'. Smth like: "Sequential" uses a jax.fori_loop over all tangent vectors to compute the diagonal of the hessian while "parallel" uses jax.vmap. "Sequential" therefore uses less memory.

File: netket/operator/_kinetic.py [hunk 2]
```diff
@@ -89,10 +96,26 @@ def logpsi_x(x):
...
+
+            def make_hess(carry, vec):
+                value = jacfwd(dlogpsi_x, vec=vec)(x)[0]
+                return carry, value
+
+            _, dp_dx2 = jax.lax.scan(
+                make_hess, None, jnp.eye(x.shape[0], dtype=x.dtype)
+            )
+            dp_dx2 = jnp.diag(dp_dx2.reshape(x.shape[0], x.shape[0]))
+        else:
```
Comment (gpescia) [inline netket/operator/_kinetic.py:114]: We could add Feenberg as:
```
elif self._method == "Feenberg":
return 0.5*jnp.sum(inverse_mass * dp_dx**2, axis=-1)
```
although this will only be valid for open systems (wave-function goes to zero at infinity) or periodic systems where the wavefunction is forced to zero gradient at the boundaries.

**Review by gpescia — COMMENTED** [review]
Looks good to me. Just a small comment below:

File: netket/operator/_kinetic.py (hunk 1 above)
Comment (jwnys) [reply netket/operator/_kinetic.py:72]: @PhilipVinc , what do you think?

Comment (PhilipVinc) [reply netket/operator/_kinetic.py:72]: I would be more inclined to think this should be two different flags.
One is the algorithm and the other is the chunk size (which does not necessarily need to be 1).

For example if you want to use `Feenberg` you might want or not to chunk it as well.
Or if you want to use Taylor mode AD or Forward mode Laplacian... those are all methods, and they might or might require chunking.

NOTE: by chunking here of course I'm referring to the chunking of the degrees of freedom. So to avoid confusion the correct name should not be chunking...

Comment (gpescia) [reply netket/operator/_kinetic.py:72]: Ok I agree with @PhilipVinc but I would only provide as algorithm: Feenberg, Taylormode, Normal
and then chunking = bool. If True: foriloop otherwise vmap.
I would not provide any algorithm with fwd-mode for the first derivative or reverse-mode for the second derivative. As far as I understand you want to use reverse mode if f:R^n -> R^m has m << n. For us we will always have m < n if not m<<n because m=1 and n = Nparticles * dim. 
Then for df: R^n -> R^(nxn) does not have m < n so you always want to use forward mode. Do I understand something wrong?

Comment (PhilipVinc) [reply netket/operator/_kinetic.py:72]: Yes indeed. The fwd fwd mode is not useful at all for our use cases
